src/controllers/Auth.ts

```diff
@@ -0,0 +1,529 @@
+    return this.query({
+      action: 'logout'
+    }, { queuable: false })
+      .then(() => {
+        this._authenticationToken = null;
+      });
+  }
+
+  /**
+   * Update credentials of the specified strategy for the current user.
+   *
+   * @see https://docs.kuzzle.io/sdk/js/7/controllers/auth/update-my-credentials
+   *
+   * @param strategy Name of the strategy to use
+   * @param credentials Updated credentials
+   * @param options Additional options
+   *    - "queuable" If true, queues the request during downtime, until connected to Kuzzle again
+   *
+   * @returns An object representing the updated credentials.
+   *    The content depends on the authentication strategy
+   */
+  updateMyCredentials (
+    strategy: string,
+    credentials: JSONObject,
+    options: { queuable?: boolean } = {}
+  ): Promise<JSONObject> {
+    return this.query({
+      strategy,
+      body: credentials,
+      action: 'updateMyCredentials'
+    }, options)
+      .then(response => response.result);
+  }
+
+  /**
+   * Update current user in Kuzzle.
+   * This route cannot update the list of associated security profiles.
+   *
+   * @see https://docs.kuzzle.io/sdk/js/7/controllers/auth/update-self
+   *
+   * @param {object} content - User custom information
+   * @param options Additional options
+   *    - "queuable" If true, queues the request during downtime, until connected to Kuzzle again
+   *
+   * @returns Currently loggued User
+   */
+  updateSelf (
+    content: JSONObject,
+    options: { queuable?: boolean } = {}
+  ): Promise<User> {
+    return this.query({
+      body: content,
+      action: 'updateSelf'
+    }, options)
+      .then(response => new User(
+        this.kuzzle,
+        response.result._id,
+        response.result._source));
+  }
+
+  /**
+   * Validate credentials of the specified strategy for the current user.
+   *
+   * @see https://docs.kuzzle.io/sdk/js/7/controllers/auth/validate-my-credentials
+   *
+   * @param strategy Name of the strategy to use
+   * @param credentials Credentials to validate
+   * @param options Additional options
+   *    - "queuable" If true, queues the request during downtime, until connected to Kuzzle again
+   */
+  validateMyCredentials (
+    strategy: string,
+    credentials: JSONObject,
+    options: { queuable?: boolean } = {}
+  ): Promise<boolean> {
+    return this.query({
+      strategy,
+      body: credentials,
+      action: 'validateMyCredentials'
+    }, options)
+      .then(response => response.result);
+  }
+
+  /**
+   * Refresh the SDK current authentication token
+   *
+   * @see https://docs.kuzzle.io/sdk/js/7/controllers/auth/refresh-token
+   *
+   * @param options Additional options
+   *    - "queuable" If true, queues the request during downtime, until connected to Kuzzle again
+   *    - "expiresIn" Expiration duration
+   *
+   * @returns The refreshed token
+   */
+  refreshToken(
+    options: { queuable?: boolean, expiresIn?: number } = {}
```

Comment:
[expiresIn](https://docs.kuzzle.io/core/2/api/controllers/auth/refresh-token/) can be either a number or a string

```suggestion
    options: { queuable?: boolean, expiresIn?: number|string } = {}
```